qiskit/pulse/commands/acquire.py

```diff
@@ -100,53 +101,96 @@ def __repr__(self):
 
     # pylint: disable=arguments-differ
     def to_instruction(self,
-                       qubits: Union[Qubit, List[Qubit]],
+                       qubit: Union[Qubit, List[Qubit]],
+                       mem_slot: Optional[MemorySlot] = None,
+                       reg_slot: Optional[RegisterSlot] = None,
                        mem_slots: Optional[Union[MemorySlot, List[MemorySlot]]] = None,
                        reg_slots: Optional[Union[RegisterSlot, List[RegisterSlot]]] = None,
                        name: Optional[str] = None) -> 'AcquireInstruction':
-        return AcquireInstruction(self, qubits, mem_slots, reg_slots, name=name)
+
+        if not isinstance(qubit, list) and not (mem_slots or reg_slots):
+            return AcquireInstruction(self, qubit, mem_slot, reg_slot, name=name)
+        else:
+            return AcquireInstruction(self, qubit, mem_slots=mem_slots,
+                                      reg_slots=reg_slots, name=name)
     # pylint: enable=arguments-differ
 
 
 class AcquireInstruction(Instruction):
-    """Pulse to acquire measurement result."""
+    """ Pulse to acquire measurement result. """
 
     def __init__(self,
```

Comment:
This is going in the right direction. I think we can still make it better. The only issue that has to be fixed is the conditional setting of the non-plural attributes (`_mem_slot`, `_reg_slot`, etc). We can kill a few birds with one stone if we use the `__init__` method to merge the possible inputs into a consistent format; in this case, it has to be the list format.

The input args are all good as is 👍 . The checks that existed before`if isinstance(acquires[0], Qubit)` and `if not (mem_slots or reg_slots)` can also stay, although mem_slots will have to be updated to `[mem_slot]` if it is provided and same for `reg_slots = [reg_slot]`. Then, I think the diff for this method will be pretty minimal.

Once we can remove the args that take a list (3 months from now, according to the deprecation policy), then we can simplify this method to the way you had it before. It might feel weird using the plural internally until then, but the users don't have to worry about the internal implementation, so we'll be able to switch from `_mem_slots` to `_mem_slot` without any deprecations or warnings.

Lmk what you think

Reply:
changed in https://github.com/Qiskit/qiskit-terra/pull/3574/commits/87b07ac3a4f3bd8e947fd6d0d3a4b08adc7781e5

Only few changes to original version of code. I agree that it looks much simpler